wandb/old/summary.py

```diff
@@ -38,13 +37,17 @@ class SummarySubDict(object):
     """
 
     def __init__(self, root=None, path=()):
+        self._path = tuple(path)
         if root is None:
             self._root = self
+            self._json_dict = {}
         else:
             self._root = root
-        self._path = tuple(path)
+            json_dict = root._json_dict
+            for k in path:
+                json_dict = json_dict[k]
+            self._json_dict = json_dict
```

Comment:
What is this doing?  Can we add a comment on what the purpose of this is?  Why not make this `self._json_dict = root.json_dict[path[-1]]`?

Reply:
That wouldn't work for nested dicts (`run.log({"a": {"b": {"c": {"loss": 0.9}}}})`). This is the same logic used in https://github.com/wandb/client/blob/master/wandb/old/summary.py#L230-L233